api/lorawan.proto

```diff
@@ -272,6 +272,8 @@ message TxSettings {
   // Time of the gateway when the uplink message was received, or when the downlink message should be transmitted.
   // For downlink, this requires the gateway to have GPS time synchronization.
   google.protobuf.Timestamp time = 11 [(gogoproto.stdtime) = true];
+  // Index of the antenna on which the uplink was received and/or downlink must be sent.
+  uint32 antenna_index = 12;
```

Comment:
`TxSettings` is specifically for transmission, from both the end device and gateway perspective. There cannot and should not be anything related to receiving here; that is `RxMetadata`.

It seems that we gradually added fields and set them along the way, while they are strictly not the transmission settings that were used by the transmitter.

I'm suggesting the following;

- Promote `gateway_channel_index` and `device_channel_index` to `UplinkMessage`
- Introduce a `TxSettings.Downlink` where we group fields that are specific for transmission from a gateway
   - `tx_power`
   - `antenna_index`
   - `invert_polarization`

Reply:
Ok done. I've rebased the commits to reflect only the newest proposal.